modules/jdbc-test/src/test/java/org/testcontainers/junit/AbstractContainerDatabaseTest.java

```diff
@@ -26,7 +31,16 @@ DataSource getDataSource(JdbcDatabaseContainer container) {
         hikariConfig.setJdbcUrl(container.getJdbcUrl());
         hikariConfig.setUsername(container.getUsername());
         hikariConfig.setPassword(container.getPassword());
+        hikariConfig.setDriverClassName(container.getDriverClassName());
```

Comment:
I found that, seemingly dependent on test order, the MariaDB driver was incorrectly being used for communicating with MySQL. This was OK normally but failed with MySQL 8.0.16. This was a strange issue that only affected my local Gradle builds, not IDE runs of the affected test, which is why I think it's likely to be test order related.

Adding this line ensures that Hikari definitely uses the MySQL driver for talking to MySQL, and resolves the issue.